torch/utils/cpp_extension.py

```diff
@@ -1056,6 +1058,18 @@ def _jit_compile(name,
                  verbose,
                  with_cuda,
                  is_python_module):
+    if with_cuda is None:
+        with_cuda = any(map(_is_cuda_file, sources))
+    with_cudnn = any(['cudnn' in f for f in extra_ldflags or []])
+    if IS_HIP_EXTENSION and (with_cuda or with_cudnn):
+        hipify_python.hipify(
+            project_directory=build_directory,
+            output_directory=build_directory,
+            includes=os.path.join(build_directory, '*'),
+            extra_files=[os.path.abspath(s) for s in sources],
+            show_detailed=verbose,
+            is_pytorch_extension=True,
+        )
```

Comment:
Maybe a better way to structure the code here is to do something similar to how TemporaryFile does it: you have an object whose destructor will clean up the temporary object. Then you can just rely the destructor to clean things up. Or better yet, have some sort of context manager (so that you can guarantee the cleanup happens in a prompt manner)

Reply:
So here is why I did come to the conclusion that leaving the generated files is better
- If something goes wrong, we might want to check them.
- If nothing goes wrong, keeping the intermediates avoids recompiles in the next run.
- To me, it seems that the context manager approach adds a lot of handing around stuff to the somewhat convoluted control flow in cpp_extension.
- Translation is only done when people use ROCm-enabled versions of PyTorch. I suspect that ROCm users will be more fond of .hip translations than, say, a non-ROCm public.

But so we can do the following:
- Wrap everything from the hipify to the ninja call in a new named context manager that will then be passed to the hipify call.
- The context manager takes a new keep_intermediates argument which defaults to False except when we are working in `load_inline` (when things happen in a temp dir, so we can leave the intermediates). This is overridable  and we don't have a new optional keep_intermediates argument.

Would that work for you?

Reply:
So, I agree that wanting to check them and recompiling them are legitimate arguments. 

So here's an alternative: why don't you name the files something distinctive, and then you can just blacklist them with gitignore. Then that woudl appease the CI?

Reply:
Or I could just switch the default to `keep_intermediates=True` and call the test with `False`...

Reply:
that would work too

Reply:
So I think we have converged. :slightly_smiling_face: 
The update now does this.